parla/task_runtime.py

```diff
@@ -291,32 +269,40 @@ def __repr__(self):
 
 
 class Task:
+    __slots__ = [
+        '_mutex', '_name', '_taskid', '_state', '_successors', '_predecessors',
+        '_req', '_assigned', '_dependent_events', '_num_blocking_predecessors',
+        '__dict__'
+    ]
```

Comment:
I have no idea what this `__slots__` thing is doing, which means it needs to be commented 😄

Reply:
ok i looked it up and that's cool and if I wrote more Python I'd probably have known. Still, please comment

Reply:
Also, what is `_num_blocking_predecessors`? I don't see it in the `__init__`.